### PR TITLE
Fix ValueTypeHelpers JDK15 Valhalla compilation failure

### PR DESCRIPTION
**Pull request by sharon-wang**
Fixes: https://github.com/eclipse/openj9/issues/9287

Split ValueTypeHelpers class name into Full and Compressed
Move VM_ValueTypeHelpers::isSubstitutable from cpp to hpp

Tested in personal build - compilation failure no longer occurring.

### CODE REVIEW

**Comment by sharon-wang** [comment]
@tajila @gacholio Please review

**Comment by gacholio** [comment]
You still need to incorporate Tobi's fixes for `_currentThread`

**Comment by tajila** [comment]
@gacholio As it is this PR will compile and function successfully. I still don't know why the `_currentThread` hasn't shown up earlier.

**Comment by tajila** [comment]
In any case, all that needs to be done is to change `_currentThread` to `currentThread` in https://github.com/eclipse/openj9/pull/9321/files#diff-1658f1fa09a27be3c43ae64c61f02510L86-L87

**Comment by gacholio** [comment]
The parameter is unused in the macros unless we're in mixed mode.

I would still like to see it fixed in this PR.

**Comment by sharon-wang** [comment]
`_currentThread` has been updated to `currentThread`

File: runtime/vm/ValueTypeHelpers.hpp [hunk 1]
```diff
@@ -45,8 +54,139 @@ class VM_ValueTypeHelpers {
...
+					Assert_VM_unreachable();
+				} /* switch */
+
+				result = fieldOffsetsNextDo(&state);
+			}
+		}
+
+	done:
+		return rc;
+	#else /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
```
Comment (tajila) [inline runtime/vm/ValueTypeHelpers.hpp:185]: nitpick: the preprocessor checks should be shifted a tab to the left

**Comment by sharon-wang** [comment]
Preprocessor checks have been un-indented

**Comment by gacholio** [comment]
Jenkins test sanity xlinuxval jdknext

**Comment by gacholio** [comment]
Jenkins test sanity zlinux jdk8

**Comment by gacholio** [comment]
Jenkins test sanity zlinux jdk8

**Comment by gacholio** [comment]
Jenkins test sanity plinux jdk8

**Pull request merged**